### PR TITLE
Issue 1865: Added profiles to enforce JDK 1.8.0-241 or 11.3 and up.

### PR DESCRIPTION
**Pull request by ghost**
Linked Issue: https://github.com/strongbox/strongbox/issues/1865

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -49,4 +49,70 @@
         <module>choco</module>
     </modules>
 
+    <profiles>
+        <profile>
+            <id>Enforce-Java-11</id>
```
Comment (carlspring) [inline pom.xml]: Could you please change this to:
```suggestion
            <id>enforce-jdk11</id>
```

Thanks! :)

File: pom.xml [hunk 2]
```diff
@@ -49,4 +49,70 @@
...
+                                    </rules>
+                                </configuration>
+                            </execution>
+                        </executions>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+        <profile>
+            <id>Enforce-Java-8</id>
```
Comment (carlspring) [inline pom.xml]: Could you please change this to:
```suggestion
            <id>enforce-jdk8</id>
```

Thanks! :)

File: pom.xml [hunk 3]
```diff
@@ -49,4 +49,70 @@
...
+            <activation>
+                <jdk>1.8</jdk>
+            </activation>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-enforcer-plugin</artifactId>
+                        <version>3.0.0-M3</version>
```
Comment (carlspring) [inline pom.xml:95]: The `version` should only be defined in the `strongbox-parent`, as we're inheriting all of our versions from there. If the plugin is not yet defined there, please add it in the `<pluginManagement/>` section. Thanks! :)

File: pom.xml [hunk 4]
```diff
@@ -49,4 +49,70 @@
...
+                                    </rules>
+                                </configuration>
+                            </execution>
+                        </executions>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+    </profiles>
 </project>
```
Comment (carlspring) [inline pom.xml:118]: Please, make sure that all source files that you add, or update, have a new line at the end of the file. Thanks! :)

**Review by carlspring — CHANGES_REQUESTED** [review]
Looks good so far, but needs some minor changes.

File: pom.xml [hunk 5]
```diff
@@ -49,4 +49,70 @@
...
+            <id>Enforce-Java-11</id>
+            <activation>
+                <jdk>11</jdk>
+            </activation>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-enforcer-plugin</artifactId>
+                        <version>3.0.0-M3</version>
```
Comment (carlspring) [inline pom.xml:63]: Pleas, remove the version, as explained.

File: pom.xml [hunk 6]
```diff
@@ -49,4 +49,70 @@
...
+                            <execution>
+                                <id>enforce-versions</id>
+                                <goals>
+                                    <goal>enforce</goal>
+                                </goals>
+                                <configuration>
+                                    <rules>
+                                        <requireJavaVersion>
+                                            <version>[11.3,)</version>
+                                            <message>Strongbox only supports JDK 11.3 and up.</message>
```
Comment (carlspring) [inline pom.xml]:
```suggestion
                                            <message>Strongbox only supports JDK 11.3, or higher.</message>
```

**Comment by ghost** [comment]
Unnecessary changes

**Pull request closed without merging**